PowerShell/ScubaGear/Modules/Providers/ExportAADProvider.psm1

```diff
@@ -1,3 +1,51 @@
+# Many of the commandlets can be replaced with direct API access, but we are starting the transition with the ones
+# below because they have slow imports that affect performance more than the others. Some commandlets are fast
+# and there is no obvoius performance advantage to using the API beyond maybe batching.
+$GraphEndpoints = @{
+    "Get-MgBetaRoleManagementDirectoryRoleEligibilityScheduleInstance" = "/beta/roleManagement/directory/roleEligibilityScheduleInstances"
+    "Get-MgBetaRoleManagementDirectoryRoleAssignmentScheduleInstance" = "/beta/roleManagement/directory/roleAssignmentScheduleInstances"
+    "Get-MgBetaIdentityGovernancePrivilegedAccessGroupEligibilityScheduleInstance" = "/beta/identityGovernance/privilegedAccess/group/eligibilityScheduleInstances"
+    "Get-MgBetaPrivilegedAccessResource" = "/beta/privilegedAccess/aadGroups/resources"
+}
+
+function Invoke-GraphDirectly {
+    param (
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $commandlet,
+
+        [System.Collections.Hashtable]
+        $queryParams
+    )
+
+    Write-Debug "Replacing Cmdlet: $commandlet"
+    try {
+        $endpoint = $GraphEndpoints[$commandlet]
+    } catch {
+        Write-Error "The commandlet $commandlet can't be used with the Invoke-GraphDirectly function yet."
+    }
+
+    if ($queryParams) {
+        <#
+          If query params are passed in, we need to augment the endpoint URI to include the params. Paperwork below.
+          We can't use string formatting easily because the graph API expects parameters
+          that are prefixed with the dollar symbol. Maybe someone smarter than me can figure that out.
+        #>
+        $q = [System.Web.HttpUtility]::ParseQueryString([string]::Empty)
+        foreach ($item in $queryParams.GetEnumerator()) {
+            $q.Add($item.Key, $item.Value)
+        }
+        $endpoint = "https://example.com" + $endpoint
```

Comment:
???

Reply:
I was also concerned when I saw this. Digging in deeper, since the endpoint ends up being the path and query concatenated together as a string, the host value itself doesn't matter. But surely there's a way to construct the endpoint string without specifying an arbitrary third-party host, even if just temporarily?

Reply:
If nothing else, you should add a comment stating that the value for host isn't actually used, i.e., no data will actually be sent to example.com. Additionally, you could specify localhost as the host, that might be at least slightly less concerning to any users reading this code.

Reply:
@adhilto Good point. I tried leave the scheme and host as empty strings and it gives the exact result we want. I will change it.